packages/reactivity/src/ref.ts

```diff
@@ -371,15 +430,31 @@ export function toRef<T extends object, K extends keyof T>(
   key: K,
   defaultValue: T[K]
 ): ToRef<Exclude<T[K], undefined>>
-export function toRef<T extends object, K extends keyof T>(
-  object: T,
-  key: K,
-  defaultValue?: T[K]
-): ToRef<T[K]> {
-  const val = object[key]
+export function toRef(
+  source: Record<string, any> | MaybeRef,
+  key?: string,
+  defaultValue?: unknown
+): Ref {
+  if (isRef(source)) {
+    return source
+  } else if (isFunction(source)) {
+    return new GetterRefImpl(source as () => unknown) as any
+  } else if (isObject(source) && isString(key)) {
```

Comment:
I think this will potentially cause a breaking change for arrays:

```js
const arr = reactive(['a', 'b'])
const c = toRef(arr, 1)
```

Previously this would have worked, but now `c` will be the whole array.